### PR TITLE
QGIS Server Enabler

### PR DESCRIPTION
**Pull request by ismailsunni**
Contents:
- Change in GeoNode core to enable [GeoNode QGIS Server App](https://github.com/ismailsunni/geonode_qgis_server) to be the backend of GeoNode
- Simple custom map view / UI for layer from QGIS Server
- Update on leaflet view, since geoext doesn't work properly with QGIS Server
- You can install the GeoNode QGIS Server App by using pip (e.g. `pip install https://github.com/ismailsunni/geonode_qgis_server/zipball/develop`)

cc @timlinux @Gustry @lucernae

### CODE REVIEW

File: geonode/templates/base.html [hunk 1]
```diff
@@ -61,8 +61,6 @@
             <li id="nav_documents">
               <a href="{% url "document_browse" %}">{% trans "Documents" %}</a>
             </li>
-           {% block extra_tab %}
```
Comment (jj0hns0n) [inline geonode/templates/base.html]: This is necessary in order to be able to override this block elsewhere.

**Comment by jj0hns0n** [comment]
Can we squash this into a single commit?

File: geonode/layers/views.py [hunk 2]
```diff
@@ -653,6 +654,58 @@ def layer_thumbnail(request, layername):
             )
 
 
+def layer_list(request):
```
Comment (jj0hns0n) [inline geonode/layers/views.py]: Shouldnt we just use the existing API for this?

File: geonode/layers/views.py [hunk 3]
```diff
@@ -653,6 +654,58 @@ def layer_thumbnail(request, layername):
...
+        else:
+            layers = Layer.objects.all()
+            result = []
+            for layer in layers:
+                result.append(layer.name)
+        result = json.dumps(result)
+        return HttpResponse(result, content_type='application/json')
+
+
+def get_layer(request, layername):
```
Comment (jj0hns0n) [inline geonode/layers/views.py]: Shouldnt we just use the existing API for this? http://demo.geonode.org/api/layers/1/

Comment (ismailsunni) [inline geonode/layers/views.py]: Hmm, I never know that there is API for it. Unfortunately, our current view, using layername (almost everywhere). We can move to use the API later (if it's ok to not do it now).

**Comment by ismailsunni** [comment]
@jj0hns0n yes, we can. I will merge this PR to kartoza repo using squash and merge feature from github, then make a PR to geonode repo with single commit only.

**Pull request merged**